filters/openpolicyagent/openpolicyagent.go

```diff
@@ -514,10 +543,84 @@ func (opa *OpenPolicyAgentInstance) Start(ctx context.Context, timeout time.Dura
 	return nil
 }
 
+// Start starts the policy engine's plugin manager and periodically trigger the plugins to download policies etc.
+func (opa *OpenPolicyAgentInstance) StartAndTriggerPlugins(ctx context.Context) error {
+	err := opa.manager.Start(ctx)
+
+	if err == nil {
+		err = withRetry(ctx, opa.triggerPlugins, []time.Duration{100 * time.Millisecond, 200 * time.Millisecond}, opa.Logger(), "trigger bundles")
+	}
+	if err == nil {
+		err = opa.verifyAllPluginsStarted()
+	}
+
+	if err != nil {
+		opa.Close(ctx)
+		return err
+	}
+	return nil
+}
+
+func withRetry(ctx context.Context, httpCall func(ctx context.Context) error, waits []time.Duration, logger logging.Logger, operation string) error {
+	var httpError download.HTTPError
+
+	err := httpCall(ctx)
+	for i, backoff := range waits {
+		isHttpError := errors.As(err, &httpError)
+		if !isHttpError || httpError.StatusCode < 500 {
```

Comment:
Should this not also include 429s?

Reply:
lets have a discussion, how likely is it that the 429 will resolve in 300ms?
but it is certainly doable

Reply:
- only add 429 to cover S3
- retries until startup limit
- grow backoff exponentially with base of 1,5